build-tools/upload-junit-to-cloud.py

```diff
@@ -1,30 +1,43 @@
 from google.cloud import bigquery
 import xml.etree.ElementTree as ET
 import argparse
+import io
+from typing import List, Optional
+import json
 
 # Uploading JUnit test results to BigQuery
-def upload(final: bool):
+def upload(final: bool, url: Optional[str], files: List[io.TextIOWrapper]):
     client = bigquery.Client()
 
     dataset_id = 'test_results'
     table_id = 'my_table'
 
-    tree = ET.parse('build/test-results/results.xml')
-    root = tree.getroot()
-    timestamp = root.attrib['timestamp']
-
     rows = []
-    for testcase in root.iter('testcase'):
-        success = len(testcase.findall('failure')) == 0
-        row = {
-            'timestamp': timestamp,
-            'testcase_final': final,
-            'testcase_name': testcase.attrib['name'],
-            'testcase_classname': testcase.attrib['classname'],
-            'testcase_time': float(testcase.attrib['time']),
-            'testcase_status_success': success
-        }
-        rows.append(row)
+    for file in files:
+        tree = ET.parse(file)
+        root = tree.getroot()
+        timestamp = root.attrib['timestamp']
+
+        for testcase in root.iter('testcase'):
+            success = len(testcase.findall('failure')) == 0
+            failures = []
+            for failure in testcase.findall('failure'):
```

Comment:
info: added information about failures to looker studio, so we can trace root problems of failures.